headers/Enums.hpp

```diff
@@ -20,14 +20,11 @@ enum class HookType {
 };
 
 //unsafe enum by design to allow binary OR
-enum ProtFlag : std::uint8_t {
-	UNSET = 0, // Value means this give no information about protection state (un-read)
-	X = 1 << 1,
-	R = 1 << 2,
-	W = 1 << 3,
-	S = 1 << 4,
-	P = 1 << 5,
-	NONE = 1 << 6 //The value equaling the linux flag PROT_UNSET (read the prot, and the prot is unset)
+enum class ProtFlag : uint8_t {
+	NONE = 0x00,
+	R = 0x01,
+	W = 0x02,
+	X = 0x04
```

Comment:
Don't remove UNSET, it's there for a reason. There is a difference between not having read the protection yet, and a variable being in an unknown state, vs having read the protection and it says there's no access.

Reply:
Ok

Reply:
Best way to solve this will be to throw an exception from protect. This way we don't end up in undefined state and prevent other problems. It's recommended way to handle errors in constructor, although I'm not a fan of exceptions that's the only good way to handle it

Reply:
No exceptions, we never throw exceptions. The flags don't have to be identical to the unix values, just map them to some unique bitflags like i do on windows, use positive numbers.